### PR TITLE
[AzureRT] Sprint 60

### PR DESCRIPTION
**Pull request by hyonholee**
1. Add additional validation for fixed vhd in Add-AzureRmVhd cmdlet
2. Add -ForceRerun parameter to Set-AzureRmVMCustomExtension, Set-AzureRmVMBginfoExtension, and Set-AzureRmAccessExtension
3. Update -VhdUri parameter from optional to mandatory (bug fix)
4. Remove GeoReplicationEnabled deprecation warning message for Get-AzureStorageAccount cmdlet
5. Fix piping issue for Remove-AzureRmExtension
6. Create on storage account for Boot diagnostics (one for each location, instead of one for each resource group)
7. Add -DiskSizeInGB paramter to Set-AzureRmVMOSDisk cmdlet
8. Show operation id and status for all POST Async cmdlets.
9. Fix Remove-AzureRmNetworkInterface issue for not throwing error when all NICs are removed.
10. Fix 'Type' output for List cmdlet
11. Remove xmlCfg contents from output format.
12. Show warning message for upcoming breaking update of Tag fix.
13. Change Set-AzureBootDiagnostics cmdlets name to Set-AzureVMBootDiagnostics (and set alias)

### CODE REVIEW

File: src/ResourceManager/Compute/Commands.Compute/Extension/AzureDiskEncryption/SetAzureDiskEncryptionExtension.cs [hunk 1]
```diff
@@ -199,7 +199,7 @@ private string GetExtensionStatusMessage()
                                                       ErrorCategory.InvalidResult,
                                                       null));
             }
-            PSVirtualMachineExtension returnedExtension = extensionResult.ToPSVirtualMachineExtension(this.ResourceGroupName);
+            PSVirtualMachineExtension returnedExtension = extensionResult.ToPSVirtualMachineExtension(this.ResourceGroupName, this.VMName);
```
Comment (markcowl) [inline src/ResourceManager/Compute/Commands.Compute/Extension/AzureDiskEncryption/SetAzureDiskEncryptionExtension.cs]: line breaks

File: src/ResourceManager/Compute/Commands.Compute/Extension/AEM/GetAzureRmVMAEMExtension.cs [hunk 2]
```diff
@@ -122,7 +122,7 @@ public override void ExecuteCmdlet()
                         this.Name).GetAwaiter().GetResult();
                 }
 
-                var returnedExtension = virtualMachineExtensionGetResponse.ToPSVirtualMachineExtension(this.ResourceGroupName);
+                var returnedExtension = virtualMachineExtensionGetResponse.ToPSVirtualMachineExtension(this.ResourceGroupName, this.VMName);
```
Comment (markcowl) [inline src/ResourceManager/Compute/Commands.Compute/Extension/AEM/GetAzureRmVMAEMExtension.cs]: line breaks

File: src/ResourceManager/Compute/Commands.Compute/Extension/CustomScript/SetAzureVMCustomScriptExtensionCommand.cs [hunk 3]
```diff
@@ -181,6 +181,13 @@ public class SetAzureVMCustomScriptExtensionCommand : VirtualMachineExtensionBas
             HelpMessage = "Set command to execute in private config.")]
         public SwitchParameter SecureExecution { get; set; }
 
+        [Parameter(
```
Comment (markcowl) [inline src/ResourceManager/Compute/Commands.Compute/Extension/CustomScript/SetAzureVMCustomScriptExtensionCommand.cs]: Does it make sense to add this to the base cmdlet? Frankly, having > 3 or 4 positional parameters is unrealistic in any case.

File: src/ResourceManager/Compute/Commands.Compute/Extension/Diagnostics/GetAzureRmVMDiagnosticsExtension.cs [hunk 4]
```diff
@@ -102,7 +102,7 @@ public override void ExecuteCmdlet()
                         this.Name).GetAwaiter().GetResult();
                 }
 
-                var returnedExtension = virtualMachineExtensionGetResponse.ToPSVirtualMachineExtension(this.ResourceGroupName);
+                var returnedExtension = virtualMachineExtensionGetResponse.ToPSVirtualMachineExtension(this.ResourceGroupName, this.VMName);
```
Comment (markcowl) [inline src/ResourceManager/Compute/Commands.Compute/Extension/Diagnostics/GetAzureRmVMDiagnosticsExtension.cs]: line breaks

**Comment by markcowl** [comment]
on demand run here: http://azuresdkci.cloudapp.net/view/1-AzurePowerShell/job/powershell-demand/829/

**Comment by markcowl** [comment]
LGTM once minor tweaks are made and the build passes.

**Comment by hovsepm** [comment]
new ondimand: http://azuresdkci.cloudapp.net/view/1-AzurePowerShell/job/powershell-demand/840/

**Pull request merged**